code/game/machinery/deployable.dm

```diff
@@ -347,3 +357,18 @@ for reference:
 			qdel(src)
 			return 1
 	return 1
+
+/obj/structure/barricade/proc/take_damage(damage)
+	health -= damage
+	if(health <= 0)
+		dismantle()
+
+/obj/structure/barricade/attack_generic(mob/M, damage, attack_message)
+	if(damage)
+		M.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+		M.do_attack_animation(src)
+		M.visible_message(SPAN_DANGER("\The [M] [attack_message] \the [src]!"))
+		playsound(loc, 'sound/effects/metalhit2.ogg', 50, 1)
+		take_damage(damage * 1.25)
+	else
+		attack_hand(M)
```

Comment:
No newline.

Reply:
> Code looks okay, but dunno about balance changes. What's copyright on new sounds? Also I personally don't really like `metal_crash.ogg`, think one of our current sounds will do better.

🤷 
According to the CM gitlab sound is licensed under the Creative Commons 3.0 BY-SA license and it's also found in the TGMC code. And yeah I'll just remove that sound as I don't think it's even being used here.

Balance wise this lets mechs punch to death things like blob, kudzu and hivemind structures. Mechs can smash light furniture (tables, racks, lockers, etc...) which... probably shouldn't block mechs in the first place? Mechs can slowly break reinforced walls at a much slower pace then what they'd manage with a diamond drill so probably wouldn't be much good if you're being actively shot at. Punching a light wall will leave a girder so it's likely still slower then using a diamond drill.